hdf5-sys/build.rs

```diff
@@ -1,3 +1,6 @@
+#![allow(clippy::must_use_candidate)]
+#![allow(clippy::option_if_let_else)]
+use std::convert::TryInto;
```

Comment:
(cosmetic, for consistency sake)

```suggestion
#![allow(clippy::option_if_let_else)]

use std::convert::TryInto;
```